### DISCUSSION TITLE
Proxy websocket with http-proxy-middleware & catch all API route

### DISCUSSION THREAD

**Discussion by birkirpalmason** [vercel/next.js#17470] [Help]
I want to set up a proxy for local development but proxying a WebSocket with http-proxy-middleware & catch all API route results in error (`invalid frame header`).

```javascript 
// pages/api/[...catch_all_dev_api].js
import { createProxyMiddleware } from "http-proxy-middleware";

export const config = {
  api: {
    bodyParser: false,
  },
}

const proxy = createProxyMiddleware({
  target:'http://localhost:3001', 
  ws: true, // proxy websockets
  changeOrigin: true,
});

export default proxy;
```

* The proxy works fine when set up with a custom server instead of dynamic API route
* Proxy works fine for `GET` requests
* example repository: https://github.com/birkirpalmason/nextjs_websocket_proxy

**Comment by dariosky** [comment]
Hi @birkirpalmason , I'm trying to solve the same problem - did you find a solution?

**Reply by birkirpalmason** [reply]
Nope unfortunately not.
